shell/platform/android/io/flutter/embedding/engine/systemchannels/SettingsChannel.java

```diff
@@ -100,4 +141,122 @@ public enum PlatformBrightness {
       this.name = name;
     }
   }
+
+  /**
+   * A FIFO queue that maintains generations of configurations that are potentially used by the
+   * Flutter application.
+   *
+   * <p>Platform configurations needed by the Flutter app (for example, text scale factor) are
+   * retrived on the platform thread, serialized and sent to the Flutter application running on the
+   * Flutter UI thread. However, configurations exposed as functions that take parameters are
+   * typically not serializable. To allow the Flutter app to access these configurations, one
+   * possible solution is to create dart bindings that allow the Flutter framework to invoke these
+   * functions via JNI synchronously. To ensure the serialized configuration and these functions
+   * represent the same set of configurations at any given time, a "generation" id is used in these
+   * synchronous calls, to keep them consistent with the serialized configuration that the Flutter
+   * app most recently received and is currently using.
+   *
+   * <p>A unique generation identifier is generated by the {@link SettingsChannel} and associated
+   * with a configuration when it sends a serialized configuration to the Flutter framework. This
+   * queue keeps different generations of configurations that could be used by the Flutter
+   * framework, and cleans up old configurations that the Flutter framework no longer uses. When the
+   * Flutter framework invokes a function to access the configuration with a generation identifier,
+   * this queue finds the configuration with that identifier and also cleans up configurations that
+   * are no longer needed.
+   */
```

Comment:
Somewhere (maybe here?) can we leave a breadcrumb to the future Android API that could replace all this and make this simpler? Just as a reminder for future readers of this code.